thrift/src/main/java/com/intellij/plugins/thrift/quickfix/ThriftCreateCustomTypeQuickFix.java

```diff
@@ -0,0 +1,86 @@
+package com.intellij.plugins.thrift.quickfix;
+
+import com.intellij.codeInsight.intention.impl.BaseIntentionAction;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.command.WriteCommandAction;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.fileEditor.FileEditorManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.plugins.thrift.ThriftLanguage;
+import com.intellij.plugins.thrift.lang.psi.ThriftCustomType;
+import com.intellij.plugins.thrift.lang.psi.ThriftTopLevelDeclaration;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiFileFactory;
+import com.intellij.psi.PsiReference;
+import com.intellij.util.IncorrectOperationException;
+import org.jetbrains.annotations.NotNull;
+
+public class ThriftCreateCustomTypeQuickFix extends BaseIntentionAction {
+
+  private final ThriftCustomType type;
+
+  public ThriftCreateCustomTypeQuickFix(ThriftCustomType key) {
+    this.type = key;
+  }
+
+  @NotNull
+  @Override
+  public String getText() {
+    return "Create type '" + type.getText() + "'";
+  }
+
+
+  @NotNull
+  @Override
+  public String getFamilyName() {
+    return "Create type";
+  }
+
+  @Override
+  public boolean isAvailable(@NotNull Project project, Editor editor, PsiFile file) {
+    return true;
+  }
+
+  @Override
+  public void invoke(@NotNull Project project, Editor editor, PsiFile file) throws IncorrectOperationException {
+    ApplicationManager.getApplication().invokeLater(() -> {
```

Comment:
I don't think there is a need for invoking later.

Reply:
This was implemented based on [this](https://plugins.jetbrains.com/docs/intellij/quick-fix.html#define-an-intention-action) article, not really know what is it for, maybe to not to lock main thread